### PR TITLE
bugfix: memory leak when call list_qsort.

### PR DESCRIPTION
**Pull request by zhuizhuhaomeng**

### CODE REVIEW

**Comment by zhuizhuhaomeng** [comment]
@afiskon waiting for your approval to run ci.

**Review by mkindahl — APPROVED** [review]
@zhuizhuhaomeng Thank you for the commit!

**Review by afiskon — APPROVED** [review]

**Comment by svenklemm** [comment]
Thanks for your contribution

**Pull request merged**
